spring-boot-starter/starter-client/spring/src/main/java/org/camunda/bpm/client/spring/impl/client/ClientFactory.java

```diff
@@ -101,6 +107,51 @@ protected void addClientRequestInterceptors(ExternalTaskClientBuilder taskClient
     requestInterceptors.forEach(taskClientBuilder::addInterceptor);
   }
 
+  protected void tryConfigureCreateTimeOrder(ExternalTaskClientBuilder builder) {
+    checkForCreateTimeMisconfiguration();
+
+    if (isUseCreateTimeEnabled()) {
+      builder.orderByCreateTime().desc();
+      return;
+    }
+
+    if (isOrderByCreateTimeEnabled()) {
+      handleOrderByCreateTimeConfig(builder);
+    }
+  }
+
+  protected void handleOrderByCreateTimeConfig(ExternalTaskClientBuilder builder) {
+    String orderByCreateTime = clientConfiguration.getOrderByCreateTime();
+
+    if (STRING_ORDER_BY_ASC_VALUE.equals(orderByCreateTime)) {
+      builder.orderByCreateTime().asc();
+      return;
+    }
+
+    if (STRING_ORDER_BY_DESC_VALUE.equals(orderByCreateTime)) {
+      builder.orderByCreateTime().desc();
+      return;
+    }
+
+    throw new IllegalArgumentException("Invalid value " + clientConfiguration.getOrderByCreateTime()
+        + ". Please use either \"asc\" or \"desc\" value for configuring \"orderByCreateTime\" on the client");
+  }
+
+  protected boolean isOrderByCreateTimeEnabled() {
+    return clientConfiguration.getOrderByCreateTime() != null;
+  }
+
+  protected boolean isUseCreateTimeEnabled() {
+    return clientConfiguration.getUseCreateTime() != null && clientConfiguration.getUseCreateTime();
+  }
+
+  protected void checkForCreateTimeMisconfiguration() {
+    if (isUseCreateTimeEnabled() && isOrderByCreateTimeEnabled()) {
+      throw new IllegalStateException(
```

Comment:
❓ @tasso94 Does `IllegalStateException` make more sense than `IllegalArgumentException` ? Or is there another exception that you think is more suitable to use instead ?

Reply:
❓ When I look at our logger class (cf. https://github.com/camunda/camunda-bpm-platform/blob/cb15a424bcceb920e1ea0b199c6759bc886cbbc4/spring-boot-starter/starter-client/spring/src/main/java/org/camunda/bpm/client/spring/impl/client/util/ClientLoggerUtil.java#L46-L56), I see that we are throwing `SpringExternalTaskClientException`. Why don't we stick to this type but introduce a new exception type like `IllegalStateException` or `IllegalArgumentException`?

Reply:
❓ Personally i'm fine with any decision, we can use `SpringExternalTaskClientException ` for all input validation erroneous cases. Would you be up for using it also for invalid sorting values? (Currently `InvalidArgumentException` is used).

Reply:
From a user's perspective, wouldn't handling exceptions be more manageable when they share the same type?

Reply:
Yeap, makes sense. I'm going to change all input validation to use the `SpringExternalTaskClientException` instead.